packages/@cosmos-building-blocks/service/test/redis.test.ts

```diff
@@ -0,0 +1,147 @@
+import { SynthUtils, expect as cdkExpect, haveResourceLike } from '@aws-cdk/assert';
+import * as cdk from '@aws-cdk/core';
+import { Redis } from '../src/index';
+import { Vpc, SubnetType, Instance, InstanceClass, InstanceType, InstanceSize, MachineImage } from '@aws-cdk/aws-ec2';
+import '@aws-cdk/assert/jest';
+
+test('Should match snapshot', () => {
+  //WHEN
+  const app = new cdk.App();
+  const stack = new cdk.Stack(app, 'aws-redis');
+  const vpc = new Vpc(stack, 'vpc', {
+    maxAzs: 3,
+    subnetConfiguration: [
+      {
+        name: 'App',
+        subnetType: SubnetType.ISOLATED,
+        cidrMask: 26,
+      },
+      {
+        name: 'Redis',
+        subnetType: SubnetType.ISOLATED,
+        cidrMask: 28,
+      },
+    ],
+  });
+  const ec2 = new Instance(stack, 'Ec2', {
+    instanceType: InstanceType.of(InstanceClass.T3, InstanceSize.LARGE),
+    machineImage: MachineImage.latestAmazonLinux(),
+    vpc,
+  });
+  // Redis cluster only allowing connections from ec2 in 'Redis' subnet
+  new Redis(stack, 'Redis', {
+    vpc,
+    connectionsAllowedFrom: ec2.connections,
+    subnetIds: vpc.selectSubnets({ subnetGroupName: 'Redis' }).subnetIds,
+  });
+  // Redis cluster allowing all connections in 'Redis' subnet
+  new Redis(stack, 'RedisOpen', {
+    vpc,
+    subnetIds: vpc.selectSubnets({ subnetGroupName: 'Redis' }).subnetIds,
+  });
+
+  // Redis cluster overwriting some of the default props
+  new Redis(stack, 'RedisOverwrite', {
+    vpc,
+    subnetIds: vpc.selectSubnets({ subnetGroupName: 'App' }).subnetIds,
+    redisProps: {
```

Comment:
this just feels weird as parent props is redis props since this the base construct for redis

Reply:
Done